hyperspy/_signals/eds.py

```diff
@@ -644,6 +644,10 @@ def get_lines_intensity(self,
                                                            FWHM_MnKa='auto')
             element, line = utils_eds._get_element_and_line(Xray_line)
             img = self.isig[window[0]:window[1]].integrate1D(-1)
+            if np.issubdtype(np.int32, np.integer):
```

Comment:
The issue is strange - we actually should be checking if `float` can be casted into whatever is the `dtype` of the data... Which is `False` if it's integer, at least for now. If numpy changes it's defaults, then probably that test (not what I actually have in the above example, but the reverse) might be more robust than just blindly converting all ints to floats though